src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/P2PProtocolHandler.cs

```diff
@@ -109,7 +109,7 @@ public override void HandleMessage(Packet msg)
                     HandleHello(Deserialize<HelloMessage>(msg.Data));
                     
                     foreach (Capability capability in
-                        _agreedCapabilities.GroupBy(c => c.ProtocolCode).Select(c => c.OrderBy(v => v.Version).Last()))
+                        _agreedCapabilities.OrderBy(c => c.ProtocolCode).GroupBy(c => c.ProtocolCode).Select(c => c.OrderBy(v => v.Version).Last()))
```

Comment:
I'm not sure if makes any difference in terms of performance under the hood (maybe it's going to optimized anyway) but I would prefer to group first and then sort.

Reply:
You're right, will change it.

Reply:
Maybe move _agreedCapabilities and _availableCapabilities from List<Capability> to SortedSet<Capability> ?